src/cryptography/hazmat/backends/utils.py

```diff
@@ -0,0 +1,38 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import absolute_import, division, print_function
+
+
+class GcmSizeValidator(object):
+    """
+    GCM may only encrypt up to 2**39 - 256 bits of plaintext, so we
+    must track the number of bytes we see.
+    """
+    _PLAINTEXT_BIT_LIMIT = 2 ** 39 - 256
+    _AAD_BIT_LIMIT = 2 ** 64 - 1
```

Comment:
@alex Does `cryptography` have a preferred position on bits vs bytes? This is a max bits value, and as such I multiply `len` by 8 below. I think this may have been a bad choice because it makes it easy to improperly check lengths, but I wanted to keep the values as close to the NIST spec as possible. (I'd prefer to change it)